### DISCUSSION TITLE
Merge inherited interface methods according to variance

### DISCUSSION THREAD

**Discussion by alrz** [dotnet/csharplang#8112] [Language Ideas]
```cs
interface ISomeInterface<in T> {
  void M(T arg);
}

interface IBase : ISomeInterface<object>;

interface IDerived : IBase, ISomeInterface<string> {
   // uncomment to resolve the error
   // void ISomeInterface<string>.M(string obj) => M(obj as object); // calls base
}
class C : IDerived {
  // error: method `M(string)` is not implemented
  public void M(object arg) { }
}
```
Could the line with explicit implementation gets generated by the compiler? (that is, if the runtime requires it)

**Comment by HaloFour** [comment]
That feels like something that perhaps the runtime should be doing rather than relying on the compiler to silently emit an explicit implementation bridge method?  Contravariant parameter types feel like a real can of worms, though.

**Reply by alrz** [reply]
> That feels like something that perhaps the runtime should be doing

I tried it in sharplap, it seems like it already passes the verification ([sharplab.io](https://sharplab.io/#v2:EYLgtghglgdgNAGxFBAfAAgNgAQH0CwAUAN5HbnYB0AxgK4DOALgPZjaxMQzUCm2AbsygATbAG0AygE8mPMJQBKtGIyhgeAXWmz5SlWp6UAwqwAOKHgCcJV/lF71jZlBFXMYCngggAPV1Hd6AEFGRksoYFpGHhAQGhZLAApYRgBmACYASmwAXmxEsgoigAYARmxi4oqADgqqyrrGysLyTJaqOiZWdhhObj5BEXFtaN1lVXUtGVHFcYMnMHMEKxtLOwdZ/XUTRf9gFChGKRCwiKiYuOoExOy8gsIikvKGssaAVgAWCvTsUpw3gDs2ABP0w5QBVQ+fEwADNsJg+B83sDqu0iphUvDkRDgV9MABOeFwhHYT7w2ogonAoEAzFldptB4UGgMFhsDiMLi8ARCUSSaZyTYTTQjQUAESgEAA5jBmEx7I4xTxIlKpRBgMsTuFItFYvFmEl+BAELQeEdTHx+TohQYptaJdLZfLqIrlbRVerNaFtecAPRKlVS2BSgCyzGEPHot3yaIorxePxe9WTdQZ7UoFssYCg9HoAV6Zuwlh4AEdszBcjGmY9sOk+K9qlVqhBSZiAYTabisfDRHXW92McDselY+QCV327W+G8qphhz34R225jx7D4XxO3XR/25/C2/PxwCvpOZ93KZhRIfl4vt2uSR9wcfj6DCZgfgDsXjkelqLWqm82yXbcjwHXtp3fZESRA1dL2/X90n/HBdxBYCV1fYk+FSURUlqPtUiqPCCMwgj4Mhfdb1/ECP2BUFkSw9cz0/Gj4XKTB4NI0limAyDQV/VdMQ+YAz0JMk7yE3cSXot9sHw7dUhXHBUkgxSgUwRTylSViviUmTZw0ldylKZF6WrR4yWKH5Sn/FdxI7RCzxfeEkNfAzcVvdCGJMopGSKSgAAsIHoPzsGNKUDUOPy2GKHxKkqRtii+X1fWwCQAAkglKdN+CsCoQGKPK8qIABfIgiBobxc2wUxwiNaIQqiZgQt6KBsAAcgAHjDYRaGWAA+VqiFIQgiuwJLsB4GBRGYOFqAq+hsE68Mep4XrSsIcrAvmlIrBhCBuWqqBar4CAGqavMQuAJhLD2xhsGAHgYQNB6oC8YRYEObAAEkJFYHhPpUHa9p4AADUp2oAWgAFVWkh2jGkMzT88N6HadBCRgHgAHd6AQZhbvVK6boEKBLEYWhjWwbcOS5AZeWwEMqxrGsAEJIZCywpW3bJ0ExdohqKEaxomqa4XURgkdEb7fv+6JLF23hQdiEM1sF5LhewabsFmzavp+9QZcBhXMsIMrtcq7a5aBqqatcY7Tq4c6CbCIn7se4sYRehA3pgD7PoAIUCnh2jUUxlnUFR5rN+apf1gHLaN9rmGAAArHgrhhobVfGyaNZmuavoD+gg5N9ao56WX5b4A6jvqlgzpap3rquO6Hqej3Xve27PqVGqeGEYPFjDibGGjwueDgbcmbLmO/rjyvQcTlO08YXqJ9MpmimnvXZ4roGF6u4MM8KQoxuLMBmGy0QQ4QKhKDv4/htGtWc816ee8Ovu1o2yrq9t2vGodi1V2bdPbew+kYdoPAfDREmvNK0Mw9DCjtDMAA8kvK4A9Q5yGHtHd+l814byZv7QOBDCGPC3tLOee8wZJ1TunUhZCKAUNjrvBOB8YBSiPqZeGiNkbpjFhLKqtANT2GwB7GAFM/IiGVDIKAUpsAY2xrjW6dhSbkxvlTXonJ+g8iGAze4jDsDsPkRADmXNsA80pqZfmTMeHi3DC3IMvQQq3QUAANSCBUHwCE3j4m3GNEwEYjFQAAF58HKIkaKpQfJM3QMiVEJcN6fQADK4FivlIsZp2hZ3VprARDijBKzWkUNGRiLTUElAgCR6giwj3KZU6pfBNF9G5IMUQ+pLD5G5rzax/jkoI3saIe6Tj5quGwO4zx0UfHAD6dgQJfA8xhOBPkaKAIYk1jidgBJ24UlpNiiAbAXtTFSkoFxdeFBdmxVKAc6gxob7Ux0W04YAoxhbBFC8ygaC6GMD1FcA0NwdmpNipgA5xZGDZKftnEW2B8miEKZcBIKtIW5LzjrCBJcgA===)) 

Either way the compiler needs to expose this support by not requiring the missing exact implementations.

**Reply by HaloFour** [reply]
Interesting, didn't know the runtime did that :D
